bodhi/server/models.py

```diff
@@ -1259,6 +1272,41 @@ def signed(self):
             return True
         return all([build.signed for build in self.builds])
 
+    @property
+    def ci_status(self):
+        """ Returns a boolean representing if all the builds associated with
+        this update are ignored or have passed their CI testing.
+
+        Returns False if at least one of the tests of the builds associated
+        with this update are queued, running or failed.
+        """
+
+        ci_status = set([build.ci_status for build in self.builds])
+        if not ci_status.difference(set([CiStatus.ignored, CiStatus.passed])):
+            return True
+        return False
+
+    @property
+    def ci_status_label(self):
+        """ Returns a label representing if all the builds associated with
+        this update are ignored or have passed their CI testing.
+
+        Returns one of: ignored, passed, failed, queued, running
+        """
+
+        ci_status = set([build.ci_status for build in self.builds])
+        if ci_status == set([CiStatus.ignored]):
+            return 'ignored'
+        if ci_status == set([CiStatus.passed]):
+            return 'passed'
+        if CiStatus.failed in ci_status:
+            return 'failed'
+        if CiStatus.running in ci_status:
+            return 'running'
+        if CiStatus.queued in ci_status:
+            return 'queued'
```

Comment:
For each of the above, you could take advantage of the Enum's built-in human readable string. I *think* you can just do ```str()``` (or maybe it's ```unicode()```? I can't remember…) on it. You can probably also just return the Enum itself and the template should just render it.